### ISSUE TITLE
UI Concept for Godot

### ISSUE THREAD

**Issue by alelepd** [godotengine/godot#2207]
Hi there,

This weekend I worked on a conceptual UI for Godot. My focus was on getting a clear and coherent design that can be flexible (fully dockable) and easy to expand in the future, and also up to date with current UX patterns on most 3d related apps. Please take a look and tell me what you think. The concept was first posted on the Godot IRC channel. Thanks to @xodene for the idea of posting the design here.

![gdoto_basic_layout_concept_01](https://cloud.githubusercontent.com/assets/6024473/8400490/301e7198-1de9-11e5-9071-e95583bb5827.png)

About the design
- Font used is Source Sans Pro, an open source font made by Adobe. This font has many weights and alternatives, including an mono spaced version, which suits perfect for scripting.
- Current icons are from Font Awesome, a set of open source icons well known in web development. My plan is to make a complete set of custom designed icons for Godot, but this could take a lot of time, so for know I sticking with this icon set, just for testing purposes.
- I took some freedom an added random icons and features, this helps me to see how the layout works.

**Update on Oct 13, 2015**
Hi everyone! I finally got time to continue my work on the UI. Here is a quick update on the interface that include some of the features you mentioned on the comments.

![gdoto_ui_concept_02](https://cloud.githubusercontent.com/assets/6024473/10529944/4143bdf6-736c-11e5-86e7-a5fb465a772e.png)

I also want to share with you a logo design concept I have been working on in parallel with the UI. This logo is my attempt to a fully coherent and modern Godot visual identity that initially starts with the UI design. Here I tried to simplify the current logo and make it more serious, solid and unique; as well as play well with different sizes and colors.

![gdoto_logotype_concept_01](https://cloud.githubusercontent.com/assets/6024473/10529960/64be9ba2-736c-11e5-8efd-8b2de84296c6.png)

As always feel free to let any feedback you want; that's best way to iterate!

**Comment by mamarilmanson** [comment]
This is an awesome UI...

You might want to add "Windows" in the menu for hiding/turning on or off some of the docks... :D

**Comment by reduz** [comment]
that looks really pretty and It's actually very coherent. The main
limitation right now is that docks are not that flexible and the main
window (2D/3D/Script/Help) can't be undocked or reorganized, but it doesn't
seem incompatible with your proposal.

The other limitation is that scene tabs, with the scenes you are currently
editing, must definitely be visible at all times somewhere at one click (no
dropdown or menu since you are moving between them very often). Any ideas
how can that be implemented?

**Comment by reduz** [comment]
i was thinking something more along the way of this (though it needs to be made pretty)

**Comment by alelepd** [comment]
@mamarilmanson that would be very useful indeed!
@reduz good point. I'm already thinking about that and I agreed, these tabs should be always visible. I will work on a new version including the mutilple scene tabs.

Glad you guys liked the concept. I will be updating this post with more views/fixes overtime, including the design of many parameters in the Inspector panel (transform, color pop ups, curves, etc).

**Comment by okamstudio** [comment]
awesome, that would be very appreciated

On Mon, Jun 29, 2015 at 12:23 AM, alelepd <notifications@github.com> wrote:

> @mamarilmanson <https://github.com/mamarilmanson> that would be very
> useful indeed!
> @reduz <https://github.com/reduz> good point. I'm already thinking about
> that and I agreed, these tabs should be always visible. I will work on a
> new version including the mutilple scene tabs.
>
> Glad you guys liked the concept. I will be updating this post with more
> views/fixes overtime, including the design of many parameters in the
> Inspector panel (transform, color pop ups, curves, etc).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/okamstudio/godot/issues/2207#issuecomment-116411355>.
>



-- 

*OkamStudio*

**Comment by xodene** [comment]
can't wait to see where this design goes! so pleasing to the eye I wouldn't mind working on my game all night

**Comment by rdcklinux** [comment]
Awesome, implement please!!!!

**Comment by KillerJaguar** [comment]
Would love to see this implemented. It would go a long to way to making Godot more appealing to new and current users.

**Comment by razvanab** [comment]
Amazing design :)

**Comment by neikeq** [comment]
This design looks amazing! I like the look in Scene and Inspector tabs.

**Comment by Calinou** [comment]
:+1: for being flat design.

**Comment by reduz** [comment]
Another issue to note is that node type icons are currently colored blue
for 2D, green for GUI, red for 3D and Yellow for animation.
Without this color coding, it's difficult to tell which version of the
nodes you are using

**Comment by alelepd** [comment]
Hey @reduz, sure, we can preserve the color coding on the icons, I think it will look good too ;)

**Comment by xodene** [comment]
you should also try to post a concept design for the project manager - would be nice to see.

**Comment by reduz** [comment]
@alelepd: awesome :)
@xodene Project manager needs some work. When you scan all the demos it becomes hell, so I was thinking of adding categories, previews and templates (that you can use to create a new project)

**Comment by reduz** [comment]
I was thinking of something more along this way for the project manager:

https://docs.unrealengine.com/latest/images/Engine/Basics/Projects/Browser/ProjectBrowserBoth.jpg

**Comment by alelepd** [comment]
@xodene indeed, I have some ideas for it.
@reduz nice! templates would be great for new users and the preview feature is very helpful too.

**Comment by macramole** [comment]
so hot

**Comment by mamarilmanson** [comment]
@reduz ~ Is it possible that in the nearest future you can implement the scripting reference as a separate offline web reference? my reference for this is how unity does it... it installs the program then when the user clicks help>scripting reference it opens the default browser to display the docs... 

I think this is good so we can remove the reference/help button in the UI, because it seems it's always a problem finding a  place or fitting it to put in the godot interface... 

:)

**Comment by mamarilmanson** [comment]
@alelepd & @reduz ~ Some ideas I have in mind...

![proposal](https://cloud.githubusercontent.com/assets/3820189/8424508/1eaf3a00-1f2e-11e5-9ea4-8a66a969d887.jpg)

**Comment by GungnirInd** [comment]
Very minor, but assuming that the 2D and 3D tabs remain (like @alelepd 's and @reduz 's examples), I would recommend that either the default (eg when creating a new scene) be changed to 2D, or the 3D tab moved to be the first; it's always stricken me as a bit odd that the editor defaults to the second tab.

**Comment by reduz** [comment]
@mamarilmanson : Scripts are good the way they are now (as a list). You usually have so many of them open that it's impossible to locate them when using tabs, being this much worse if you mix them with the scenes

**Comment by draxdeveloper** [comment]
yeah, really liked de design... There is some people that when come to
godot goes with a: wel... i don't know... Deisgn help with that part.
But speaking of UI... there is a bug in the actual one, that i can't dock
more than 2 tabs in the same docking area

2015-06-30 9:56 GMT-03:00 Juan Linietsky <notifications@github.com>:

> @mamarilmanson <https://github.com/mamarilmanson> : Scripts are good the
> way they are now (as a list). You usually have so many of them open that
> it's impossible to locate them when using tabs, being this much worse if
> you mix them with the scenes
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/okamstudio/godot/issues/2207#issuecomment-117167341>.
>



-- 
David Aguiar de Aquino Paiva

**Comment by draxdeveloper** [comment]
Ah! would be very useful to undock the output, even more if we can undock
stuff and put in a separeted window in the future... You can double the
usefulness of this with people that have two screens... (so one to output
and other to thre rest)

2015-06-30 10:30 GMT-03:00 David Paiva <draxdeveloper@gmail.com>:

> yeah, really liked de design... There is some people that when come to
> godot goes with a: wel... i don't know... Deisgn help with that part.
> But speaking of UI... there is a bug in the actual one, that i can't dock
> more than 2 tabs in the same docking area
>
> 2015-06-30 9:56 GMT-03:00 Juan Linietsky <notifications@github.com>:
>
>> @mamarilmanson <https://github.com/mamarilmanson> : Scripts are good the
>> way they are now (as a list). You usually have so many of them open that
>> it's impossible to locate them when using tabs, being this much worse if
>> you mix them with the scenes
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/okamstudio/godot/issues/2207#issuecomment-117167341>.
>>
>
>
>
> --
> David Aguiar de Aquino Paiva
>



-- 
David Aguiar de Aquino Paiva

**Comment by OlexiyKravchuk** [comment]
I very much look forward to the steps towards the interface the same as in a blender to all the parameters of the position and the configuration so that you can adjust as you like in any order and configuration.

**Comment by alelepd** [comment]
@mamarilmanson I agreed on that, it would be great if the reference is online, using a menu button for the help section is also a better ux pattern, actually that's where normally is on other apps.
+1 for the X icon on the scene tabs, that's the way I think of it (same as in chrome). The 3d/2d switch is insteresting, this is way unity works, in our case, that icon should be on the left corners because 2d and 3d views have different options, although maybe there's a way to standardize it.

The idea for the scripts in the same bar as the scenes looks interesting although right now I'm liking more the reduz approach (switching between scenes and scripting views) for this particular feature. If you have many scenes and scripting tabs open it could be a little mess. That said, your idea works perfect when having few files opens.

I will think deeply on this by and share here what I come up this weekend.

@GungnirInd Yup, I think 3D should be first and always the default option, it should be in perspective camera by default too (easy to understand that it's a 3d view).

**Comment by reduz** [comment]
@alelepd The built-in help is useful for many reasons. First one is that it generates this help from the currently running version of the engine when compiling, so if you are using a development release, this documentation is always up to date (no function or class will be missing). The other reason for it is that, even though this is not implemented right now, the plan is that properties being edited can show tooltips extracted directly from the internal docs to make it more obvious what everything does

**Comment by kubecz3k** [comment]
@alelepd Another reason to have build in docs is the fact they work offline :)

**Comment by xodene** [comment]
It's actually a smart idea to have the documentation within the editor. The nice thing about Godot's editor is that workflow is always focused on the editor and that's really nice. Others benefits for in-editor docs have already been explained. I don't see how opening up a web browser for the documentation is a better UX experience, I'd say it's the opposite.

**Comment by draxdeveloper** [comment]
agreed

2015-06-30 12:22 GMT-03:00 Maximillian <notifications@github.com>:

> It's actually a smart idea to have the documentation within the editor.
> The nice thing about Godot's editor is that workflow is always focused on
> the editor and that's really nice. Others benefits for in-editor docs have
> already been explained. I don't see how opening up a web browser for the
> documentation is a better UX experience, I'd say it's the opposite.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/okamstudio/godot/issues/2207#issuecomment-117226487>.
>



-- 
David Aguiar de Aquino Paiva

**Comment by KillerJaguar** [comment]
One of my favorite things about GameMaker is its built-in documentation and its ease of access. It goes a very long way and is a major contributing reason to why I like Godot.

**Comment by OlexiyKravchuk** [comment]
Yes, built-in documentation is 100% that thing which I dream to have in any software I should mention if it was not in Godot, it would be a terrible pain for me, I love it, even despite the fact that it is now not complete, but it really is very necessary so far, and there is a benefit here and now, it's all in one, without the Internet and unnecessary searches!

**Comment by mohaalak** [comment]
Offline help is so good,  I love it so much, it helps me alot.

**Comment by mamarilmanson** [comment]
@alelepd ~ Now... need to create a UI creative team... XD

**Comment by mamarilmanson** [comment]
@reduz ~ sir... also maybe... if you can start cooking the d&d support for godot while we cook this UI brainstorming... uhm please... :)

**Comment by draxdeveloper** [comment]
since we are here... it would be a good idea the possibility  to change
grid color, since the viewport border confuses with the grid lines...

2015-06-30 17:32 GMT-03:00 Manson Mamaril <notifications@github.com>:

> @reduz <https://github.com/reduz> ~ sir... also maybe... if you can start
> cooking the d&d support for godot while we cook this UI brainstorming...
> uhm please... :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/okamstudio/godot/issues/2207#issuecomment-117333062>.
>



-- 
David Aguiar de Aquino Paiva

**Comment by alelepd** [comment]
These are great arguments! It makes me rethink about the reasons why I like the online reference and I think it's only because is easy to share and access it from any device; aside from that, the offline help definitely wins. For online help we have the wiki.

@xodene what I mean by a better UX pattern is that is quite common to find the Help or reference docs from a menu instead of a panel, which are normally used to do actions or editing. Tha said, I think Godot help is fine the way it is :D. I do think there's room for improvement though (visually speaking).

@mamarilmanson haha, good point!

@draxdeveloper you mean grid color on the concept (white) or the current color on Godot (green)?

**Comment by mamarilmanson** [comment]
@alelepd ~ That's a good keyword to use... "Visually Speaking..." that will better explain that we only want to change how it looks and how it interacts with us... not how it works... :D

**Comment by vkbsb** [comment]
this looks great.. we should totally get this in godot.. !!

**Comment by TeddyDD** [comment]
Wow! My eyes of artist are happy to see that nice concept :D 

About Project manager @reduz What about sorting projects by tags?

**Comment by mamarilmanson** [comment]
@reduz ~ It's really time to please designers... XD

**Comment by trollworkout** [comment]
custom skins and docking tabs would be nice. i like current skin better than the proposed above skin but i like the ui structure very useful.

**Comment by reduz** [comment]
it should be relatively easy to make custom skins, there should just be
better documentation on how to do it

**Comment by trollworkout** [comment]
u guys must keep 2d 3d  scripts an help is much better system than tabs suggested above . also i like how you focus on one scene at a time and how resources can load anything that you quickly wanna access (common container)

 would be nice to be able to quickly edit /create images ie like ms paint in 2D view just basic tools select, pencil, brush, eraser . and edit/create 3D models in the 3D view. and help needs more text . the online wiki has more info. i want something like manpages. ideally these 3 changes will enable me to never have to leave Godot

**Comment by mamarilmanson** [comment]
D&D would be the greatest first step… if you ask me… :)

**Comment by reduz** [comment]
I'm trying to use multiple scene editing for Okam projects and have to admit it's kind of confusing for me, as having to select scene and script from different lists is difficult, and having to read the opened scene tabs for the scene i'm editing is not as straightforward either. 

Going from the scene I have in my mind to a tab requires reading and that's kind of a slow process. I tried to understand why this is so efficient in web browers, and then realized it's obviously the website icons doing the work and not the text. Having the scene disassociated from the script, when in most cases it's related is also kind of annoying. 

I've been trying to figure out ways to make this more efficient and realized we can leverage the idea of using scene thumbnails for switching (since they are being generated anyway, and generation could easily become more "real-time" too without much of an issue. 

So, how about something like this, showing scene thumbnails AND scripts related to the scene?

![mse](https://cloud.githubusercontent.com/assets/6265307/8875098/eb23f438-31ec-11e5-8832-416c2628443f.png)
 
It's easy to find a scene and related scripts this way.  Another problem I usually face is the disconnect from the scene tree and property editor . they are in opposide sides of the screen now by default, so it's not so obvious that they connect.  As such, I was considering moving scene tree to the right as it was before, and keeping the left for opened scenes and filesystem.. and do away with resources dock (integrate a similar functionality as a menu in the property editor)

**Comment by slapin** [comment]
Well, from my perspective, if you have a small amount of scripts (under 20) it is quite easy to
display scripts, referenced from scene on top of the list and in different color.
However, that won't help when you have > 100 of scripts most of whiach are not referenced from node tree by default. To make this work I'd appreciate being able to group scripts into packages (the python way) being presented as subdirectories.

**Comment by zhagsenkk** [comment]
My suggestion.And i think we should allow to use right button.
![ss](https://cloud.githubusercontent.com/assets/10054853/8887792/89c4a2f4-32c4-11e5-9562-ba4bec390f5c.png)

**Comment by TeddyDD** [comment]
@zhagsenkk I would add pop-up thumbnails when mouse hover over tabs :+1:

**Comment by kubecz3k** [comment]
@zhagsenkk Hmm I think @reduz want to utilize scene thumbnails to perform this functionality.. as he said finding tabs with text is not as fast and easy for our brain as finding pictures..  The problem is how to use them in the way that will look good and still will be functional (don't eat half of the screen I guess).

**Comment by zhagsenkk** [comment]
@kubecz3k I think we should use keyboard shortcuts to change scene.Like Alt+Tab in Windows 7 or 8. Show us a diagram.

**Comment by Calinou** [comment]
`Ctrl+Tab` is used by most browsers to switch tabs. How about this shortcut switches scenes? `Ctrl+Shift+Tab` would inverse the order of switching, just like in browsers.

**Comment by reduz** [comment]
Added root node icons on the tabs, this seems to considerably help. I will
try to toy around with adding previews and a way to see the scripts
associated too

**Comment by trollworkout** [comment]
there are few minor annoyances with the ui but as it stands i think is awesome and doesn't need heavy restructuring. someone mentioned keyboard shortcuts. 

how about 1 = 2D, 2 = 3D , 3 = Script, 4=Help  or Ctrl+1, Ctrl+2 ...

would be nice to have an manually assignable keyboard shortcuts for various actions/panes and saveable/loadable keyboard shortcut config file.

**Comment by reduz** [comment]
you have F2,F3,F4,F1 for 2D, 3D, Script and Help respectively

**Comment by trollworkout** [comment]
i didn't know that. that's good enough :+1:

**Comment by draxdeveloper** [comment]
The scenes can have color tabs and/or icons (the user can put a personalized icon).
This would help since the user would choose the best color/icon that fit the concept to him (i could put green tabs to character and red to enemies scenes, for example. Another user would prefer a different approach) 
The scripts, i liked the @slapin packages idea (even if i don't grasp the concept entirely).
Another solution it's one that i mentioned before, using tags on the scripts.
So i could filter, group and sort using those tags.

**Comment by reduz** [comment]
I ended up adding a script button to the tabs, and I think it worked pretty
well

**Comment by draxdeveloper** [comment]
script button, how it will work?

**Comment by rdcklinux** [comment]
if you click the button then go to scene's script

**Comment by shackra** [comment]
> you have F2,F3,F4,F1 for 2D, 3D, Script and Help respectively

Why not deleting that and use <kbd>C-x</kbd> <kbd>b</kbd> instead... :P

**Comment by mamarilmanson** [comment]
what character is that? :)

**Comment by draxdeveloper** [comment]
So, i looked the script button but i still think that would be good script tags to sort and filter the scripts, and also would be good a option to personalize the icons of the nodes, i have a lot of scenes that starts with a node 2d, so a lot of then would be just a blue ball.

**Comment by BBric** [comment]
Icons for the scripts because the asterisk is often hidden.
![scripts icons](http://www.pasteall.org/pic/show.php?id=91434)

**Comment by draxdeveloper** [comment]
Ok, i at this moment i have 30 scripts and this is a small project that still have stuff to add...
That said, it's becoming harder to find my scripts, not because i can't find it but because i can't organize my list by groups, so i can't work on just one thing... at the moment i can't handle this, but think in 100 or more scripts (and i can easily hit this in a more complex project) ok, i can use the open shortcut, but still is not practical or at least this make the side list less useful for those cases...

**Comment by xodene** [comment]
The script editor needs to show the directory tree for scripts, then it would be much more organized to find certain scripts. Similar to how this is done in many IDEs.

**Comment by shackra** [comment]
I say that this is better tacklet adding a shortcut for showing a modal window which lists all your scripts
and you can narrow the candidates by writing in a "line edit". Something like Helm for Emacs, but adapted for Godot:

![](https://tuhdo.github.io/static/helm_projectile.gif)

Heck, I even suggest this solution for opened scripts too (a shortcut for switching between open scripts using
the same system).

El martes 11 de agosto del 2015 a las 1447 horas, Maximillian escribió:

> The script editor needs to show the directory tree for scripts, then it would be much more organized to find certain scripts. Similar to how this is done in many IDEs.
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/okamstudio/godot/issues/2207#issuecomment-130065336

-- 
Pax et bonum.
 Jorge Araya Navarro.
 ES: Diseñador Publicitario, Programador Python y colaborador en Parabola GNU/Linux-libre
 EN: Ads Designer, Python programmer and contributor in Parabola GNU/Linux-libre
 EO: Anonco grafikisto, Pitino programalingvo programisto kai kontribuanto en Parabola GNU/Linux-libre
https://es.gravatar.com/shackra

**Comment by draxdeveloper** [comment]
@alelepd
>  you mean grid color on the concept (white) or the current color on Godot (green)?

The actual green, the grid color and the color of the "border" of the viewport are the same

**Comment by reduz** [comment]
I use the script open shortcut all the time (alt-ctrl-o), but i'm not
really sure if there are better ways to tacke this problem..
i thought about making the script list sortable by recent-usage as an
option, but when I develop godot (which has thousands of C/C++ files), only
the locator really works.

**Comment by draxdeveloper** [comment]
i still think that putting tags on the script that can be grouped, filtered and sorted would solve this.

**Comment by draxdeveloper** [comment]
There is a modding tool called mod organizer that do this with their mods, i am not in my home at the moment, when i get there i can't show a printscreen of it.

**Comment by trollworkout** [comment]
one really simple solution is ability to tag scripts files using color kinda like mac or google does  then have some tiny toggle-able buttons one for each color that can show or hide scripts that have been tagged. and have like 6 colors or 8.

**Comment by trollworkout** [comment]
i find color coding to be most efficient way to visually organize something . i use it all the time in g drive or on my mac. is much faster than reading text tags.

**Comment by draxdeveloper** [comment]
text tags would not be used to read, but to group and filter.
grouping would be like this
tag A
 yadayada
 yadayada
 yadayada
tag B
 yadayada
 yadayada
 yadayada
tag C
 yadayada
 yadayada
 yadayada

and filtering would just show scripts with the said tags, so you can short a list with hundreds of scripts

**Comment by draxdeveloper** [comment]
But why not both things, you can say that a tag have a color, i thing github is that way

**Comment by trollworkout** [comment]
i just suggested color tagging as it's the easiest to implement. also i'd rather filter it by showing and not showing when tag is enabled rather than as a list. 

another way of doing it is have a small live search box that filters all script files that contain that text

file1.gd
bigscript.gd
blahblah.gd

type "b" you get second two script showing, type "bi" you get only bigscript showing. etc. then you can create your own system by naming scripts in a certain way. i've seen this being done and rather successfully in geck fallout 3 modding software for example which has millions of entries and easily filterable.

for this system all you need is add an input box that you can type in.

**Comment by draxdeveloper** [comment]
here:
http://snag.gy/LpIIY.jpg

**Comment by draxdeveloper** [comment]
you can edit the tags on this list, for example, realism is a tag that i have created.

**Comment by kubecz3k** [comment]
Hmm I'm wondering why in new ui (nightlies) opened scripts in the script list are not assigned to scene... That way how it was working in vers <= 1.1. It was very natural and 'godotish' way to group scripts. For me it was very comfortable that when changing scene another set of tabs with scripts were prepared. I Usually work with some of scripts that are very much linked to current scene as well as to themselves. And I very often was changing scene because I wanted to load 'different group' of scripts to the list. It would be even more awesome now, when scenes can be changed by visually clicking on the tab.

**Comment by mamarilmanson** [comment]
@draxdeveloper @alelepd ~ Hello... :)

**Comment by draxdeveloper** [comment]
@manson Mamaril hi

2015-08-12 22:32 GMT-03:00 Manson Mamaril <notifications@github.com>:

> @draxdeveloper <https://github.com/draxdeveloper> @alelepd
> <https://github.com/alelepd> ~ Hello... :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/okamstudio/godot/issues/2207#issuecomment-130499139>.
>



-- 
David Aguiar de Aquino Paiva
Dive sistemas

**Comment by alelepd** [comment]
Hi!
Hey guys, sorry for been inactive since my last post. I'm actually really busy these days and it has been quite difficult to get the time to work on the UI.
But I read everything you are writing here. I hope to get some free time in the incoming weeks and continue my work with the UI :D

**Comment by mamarilmanson** [comment]
ahaha XD

**Comment by trond** [comment]
I think @trollworkout has a great suggestion with the live search box for finding scripts. Visual studio has this, and since I discovered that I have been using it all the time - between that and "go to definition", I barely ever use the file explorer any more, although it is of course useful for getting to know new projects.

**Comment by trollworkout** [comment]
2.0 UI is much much better thank you <3

**Comment by alelepd** [comment]
Update time! (on initial post).

**Comment by reduz** [comment]
i'm finishing the last features for 2.0, will hopefully integrate the new
UI on the weekend.

**Comment by xodene** [comment]
Again, this is incredible work. I can't wait to work in this environment. 2.0 stable is going to be amazing.

**Comment by draxdeveloper** [comment]
godot is growing everyday, at least enter one new member each weeek on the
godot brasil group

2015-10-15 20:26 GMT-03:00 Maximillian <notifications@github.com>:

> Again, this is incredible work. I can't wait to work in this environment.
> 2.0 stable is going to be amazing.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/okamstudio/godot/issues/2207#issuecomment-148550492>.
>



-- 
David Aguiar de Aquino Paiva
Dive sistemas

**Comment by alelepd** [comment]
@reduz that's great!
Any chance we can get it with support for retina displays in OSX?
I really think that would be huge benefit for those using hi-resolution screens.

**Comment by reduz** [comment]
I'm too poor for a retina display, so that will have to wait until I can
save enough money or the community donations exist and can gather enough
money for one :P

**Comment by silverkorn** [comment]
@reduz, lot of people have suggested donation portals (Patreon, Kickstarter, etc.). 
You should take few minutes to launch one so we can invest in Godot's development or just for the appreciation.

Are donations through those services something complicated to manage with your country's law?

**Comment by okamstudio** [comment]
We are working with Conservancy on setting up a donation system. It is a
lot easier for the project to rely on a foundation when looking for
commercial, education or government sponsors.

On Fri, Oct 16, 2015 at 12:40 PM, Juan Linietsky <notifications@github.com>
wrote:

> I'm too poor for a retina display, so that will have to wait until I can
> save enough money or the community donations exist and can gather enough
> money for one :P
>
>
> On Fri, Oct 16, 2015 at 11:37 AM, alelepd <notifications@github.com>
> wrote:
>
> > @reduz <https://github.com/reduz> that's great!
> > Any chance we can get it with support for retina displays in OSX?
> > I really think that would be huge benefit for those using hi-resolution
> > screens.
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/okamstudio/godot/issues/2207#issuecomment-148734387
> >.
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/okamstudio/godot/issues/2207#issuecomment-148749784>.
>



-- 

*OkamStudio*

**Comment by trollworkout** [comment]
guys just start a crowdsource campaign.  people will like this project and you might be able to get a few things you need.

**Comment by MarianoGnu** [comment]
I'm a little scared with screenshot resolution... 2.880px × 1.860px
If the controls are not scaled down by half my screen wont have enought space for everithyng, i'l end up working in a 32x32 canvas editor ^ ^!

Can't wait to try this out :D

**Comment by alelepd** [comment]
@MarianoGnu hehe, don't worry about that, I'm designing on a retina screen, which double the resolution. The non retina equivalent is half the size, a 1440px width screen ;)

**Comment by danilojr** [comment]
Hi,
 Here's a suggestion for the logo updating. It is just a simplification of the original (which is already pretty cool).

![logo-suggestion](https://cloud.githubusercontent.com/assets/13174433/10557847/141fde8c-7492-11e5-88ca-67e237b37cab.jpeg)

**Comment by Calinou** [comment]
The current logo looks like this:

![Godot logo](https://raw.githubusercontent.com/okamstudio/godot/master/logo.png)

**Comment by reduz** [comment]
Andrea, the logo designer, has expressed desire to simplify the design a
little but said she will wait until 3.0

**Comment by BBric** [comment]
The current logo is cute, ludic and suggest gears and other mechanisms. It's not necessarily a good idea to change that.

**Comment by MarianoGnu** [comment]
besides, that looks like an hippo.... i like the G-bot much more :)

**Comment by alelepd** [comment]
@danilojr great! My approach was more in the "serious warrior" mood, trying to create something that clearly states "I am powerful and amazing" in comparison with "I am funny and friendly" statement of the current logo. Although keeping the mascot/warrior/robot idea is totally fine too, maybe we can make something that states "I am powerful and friendly". Your approach looks nice, I think the version without the nose is the better.

I think a revamp of Godot visual identity is needed in order to get more people involve in the community, people need to see that Godot can be more than just a friendly game engine but a tool to create amazing gaming experiences. In my research, Godot engine visual identity is quite below the bar when is compared with other engines like Unreal, Cryengine or Unity. Godot have the potential to become one of the "big names" in game creation, same way Blender is doing it on animation, but of course that will only happen if the goals are set towards that.

In the other side, it would be hard to make a drastic change that everyone likes at first sight because the current logo is already placed in the community's memory, so most people will think its an unnecessary change, even when its not. This is actually very common in most open-source projects, where design is usually taken as something with low priority, commonly mistaken as a matter that only affects aesthetics. What normally happens in this arena is that changes happen slowly through the years.

**Comment by shackra** [comment]
> maybe we can make something that states "I am powerful and friendly". Your approach looks nice, I think the version without the nose is the better.


Without a nose!?

![growing-a-nose-makes-you-stronger-dbz-logic](https://cloud.githubusercontent.com/assets/1055216/10562178/577dabfe-750b-11e5-8bd9-550b159eb7c6.jpg)

**Comment by alelepd** [comment]
@shackra Ok ;) I will publish new updates through the forum.

**Comment by draxdeveloper** [comment]
the noseless one don't seems a robot, it's seems some kind of a orc.

2015-10-17 23:12 GMT-03:00 Jorge Araya Navarro <notifications@github.com>:

> maybe we can make something that states "I am powerful and friendly". Your
> approach looks nice, I think the version without the nose is the better.
>
> Without a nose!?
>
> [image: growing-a-nose-makes-you-stronger-dbz-logic]
> <https://cloud.githubusercontent.com/assets/1055216/10562178/577dabfe-750b-11e5-8bd9-550b159eb7c6.jpg>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/okamstudio/godot/issues/2207#issuecomment-148968953>.
>



-- 
David Aguiar de Aquino Paiva
Dive sistemas

**Comment by trollworkout** [comment]
i cant handle the no eyeballs white eyes that's my main concern :)

**Comment by silverkorn** [comment]
> "I am powerful and friendly".

Just make the Godot mascot becoming mad for 3.0 :laughing: 

![godot-mad](https://cloud.githubusercontent.com/assets/2325424/10580458/94f1ca56-764b-11e5-8d08-698971d7cb96.png)

**Comment by trollworkout** [comment]
nice

**Comment by ssavalot** [comment]
I made a dark flat ui layout for Godot engine. 
The color sets based on alelepd (cheers) designs.

Download:
https://github.com/ssavalot/godot_flat_ui

Merry Christmas!

**Comment by reduz** [comment]
this looks really good! let's use it for 2.1

**Comment by shackra** [comment]
Awesome! :open_mouth: 

Thank you and Merry Christmas, @ssavalot !

**Comment by reduz** [comment]
2.1 will have many improvements, such as drag and drop, context menus, etc.

**Comment by Calinou** [comment]
What about using `#547799` as color code for comments in the script editor? It is more readable than the current brown color, and fits the theme more.

**Comment by trollworkout** [comment]
hey guys. I was trying to change the godot UI color to something else at it seems the colors in settings only affects certain parts of the UI namely Text Editor. I wish I could change the color of the entire UI kinda like say Gimp or Ableton Live can. I find the background color too blue I wanted it to be more gray maybe warm gray.

**Comment by toger5** [comment]
Hey,
first of all I really love the style of the UI concept It looks Great and professional.
I also want to make an proposal about the UI. Because I think having one big UI element for rhe three modes (3d/2d/scripting) doesn't make that much sense anymore because the script editor isn't as scene dependent as it was. I like the Idea of @mamarilmanson. Because you usually only have 3D Or 2D scenes it would make a lot of sense to make this a toggle control.  So my idea is to make the scene tabs on top and implement the 3 different views (3d/2d/scripting)  into the task bar of the main window. 
Like so:
![godotuiproposal](https://cloud.githubusercontent.com/assets/16718859/13373346/231d45a4-dd66-11e5-8d59-5ac8da80f374.png)
compared to:
![godotuiproposal_oldversion](https://cloud.githubusercontent.com/assets/16718859/13373372/5153b7b8-dd67-11e5-902b-84eb6cfe4a7d.png)
This would have the advantage that there is less space occupied for task bars -> you get more space.  See Issue:  #3779 Godot UI improvements [Mockups]

And I think It wound make a lot of sense to toggle between scripting and scene editing. 

I also wanted to ask if anybody can give an update about the priority of new UI design and when it may be implemented ?

I just want to mention (although I know its not the right place, but it was discussed here too ) that I really like the Logo of the [DarkDot](http://www.godotengine.org/qa/46/darkdot-dark-theme-for-godot-q%26a) theme of the Q&A:

![image](https://cloud.githubusercontent.com/assets/16718859/13373437/a45ba5a0-dd68-11e5-87fa-209d2c8d608b.png)
It also fits very well to this theme

**Comment by reduz** [comment]
Please remember that in Godot, both 2D and 3D engines are separated, so you
can't really expect their toolbars to be the same or have a single button
to switch between 2D and 3D. I know it looks prettier but it's not that
usable this way
.

**Comment by alelepd** [comment]
It is awesome to see that some folks have come with such great ideas for the UI! Never thought that my initial post would had so many feedback comments. To me, it means that many people in Godot community actually cares about design and look foward to see more improvements in that regard.

I really hope that Godot development could add more priority into the User Interface design. Current Godot UI could benefit a lot from some simple design changes. There is room for improvement anywhere, from the website (yes, is new but it can be much better) to the user onboarding of the app (just look how many steps you need to start a simple project), the logo and even the icon (no more gray background please).

It is a huge mistake to think that just because something is working "ok" it can't be much better.

There have been some great progress with Godot as a Game Engine this last year, but unfortunately the progress on UI is incredible low.

Last year I took the initiative to work on this because back then I already saw that Godot UI needed help, 9 months later and I now see that it should be a no-brainer choice.

**Comment by reduz** [comment]
> I really hope that Godot development could add more priority into the User Interface design.

2.0 Changed the user interface **a lot**, pretty much every aspect of the user interface was improved. 
2.1 Will continue improving it, including a new and more user friendly project manager. 

We haven't switched to a more modern flat-looking theme though, current one is too new, so we'll wait a bit and work un usability first.

**Comment by MarianoGnu** [comment]
There's no comments anywhere about this, but since color picker was added, there's a way to add default shaders and materials to default theme, this will need some changes in code (maybe add a ShaderMaterialStyleBox class) but it's posible and may allow to scale controls without losing quality.

**Comment by LaBlazer** [comment]
We should really implement vector graphics for UI. I cant even say how i like this design, it should be really implemented. I was working with shader graphs and they only look good on 1:1 scale, if you zoom in or zoom out them they become very pixelated and non-readable. I agree that usability is a priority, but UI shouldn't be overlooked too...

**Comment by silverkorn** [comment]
> We should really implement vector graphics for UI.

[nanosvg](https://github.com/memononen/nanosvg) & [nanovg](https://github.com/memononen/nanovg) could be good candidates, cross-platform and really permissive licence (zlib). :smiley:

**Comment by reduz** [comment]
nope, Opengl is not designed to draw this stuff in real time. At much it
can be baked to a mesh but not modified much in real-time.
Flash uses vector and renders using CPU, but for us this is not possible.
2D Meshes will be introduced in Godot 3.0, so for now we will have to wait.

In any case, we can simply use a texure in higher resolution and zoom will
look OK, it's not too bad.

**Comment by alelepd** [comment]
I'm not completely sure how, but as I understand, Blender UI is done with pure OpenGL. You guys may want to check that.

**Comment by toger5** [comment]
A while ago @ssavalot posted a really nice looking theme. I wanted to adapt it a little more to make It look a little more like the one alepd designed.
I changed some .png's and some code (for example to style the top bar without the button hover stylbox and to make the tabs float to the left.)
Here are some screenshots just to give you an impression how it looks so far.
![theme almost done3](https://cloud.githubusercontent.com/assets/16718859/14230135/2d2fe338-f94c-11e5-93e6-578c155d57db.png)
![theme almost done](https://cloud.githubusercontent.com/assets/16718859/14230134/2d2fb94e-f94c-11e5-8a26-9543d2c4ac9c.png)
![theme almost done2](https://cloud.githubusercontent.com/assets/16718859/14230136/2d34f490-f94c-11e5-9727-e8dd6d4da8b6.png)

**Comment by reduz** [comment]
Nice.. Will definitely want to merge it for 2.1

**Comment by toger5** [comment]
I wanted to remove the padding of the tab containers on the left and on the right (so that the viewport has contact to the background as in the UI concept of @alelepd ...). But I had no success. Can anybofdy give me a hint where to finde it (code line and document). I guess it have to be in editor_node.cpp or default_theme.cpp or do I have to change the TabContainer class ?

**Comment by KioriSun** [comment]
About the logo, i think the helmet logo alelepd created is eons better than the current one, far simpler, and more compatible with the many scenarios a logo has to show up in.
Also, more engaging as a tool symbol.

**Comment by Dillybob92** [comment]
Holy shit that UI is amazing. Good work OP.

**Comment by Geequlim** [comment]
It is really cool!

I did some syntax schemes. I think this one is fit for this editor theme.
![chester](https://github.com/Geequlim/godot-code/blob/master/assets/images/styles/chester.png)

The syntax schemes can be found [here](https://github.com/Geequlim/godot-code/tree/master/styles/syntax)

**Comment by kubecz3k** [comment]
Adding 2.1 milestone to consider, since @reduz mentioned that this could be merged in 2.1

**Comment by neikeq** [comment]
@Geequlim Amazing color schemes! It may be good to provide `.tet` files once #4296 is merged :)

**Comment by reduz** [comment]
merged

**Comment by trollworkout** [comment]
I personally prefer warmer colours and that green is jarring   but it does make the code pop out : D

**Comment by Dillybob92** [comment]
@trollworkout we can choose I believe so it'll be okay

**Comment by trollworkout** [comment]
Hey guys I love @toger5  's layout. I had a few ideas and instead of making a new post I decided to just edit a few things based on his layout. 

- Removed all the panel tab titles. You can clearly see what a tab does without the title. While it may look nice is a waste of space. 
- Centre panels   now get the ability to maximize to full window. This will affect 2D, 3D, Script as well as Output, Debugger and Animation. Filesystem Browser should also be able to do this.
 
![idea](https://cloud.githubusercontent.com/assets/6987027/15561237/99bd4970-22c1-11e6-8d3d-174962b72526.png)

Also proposing panel icons to be reduced in size.
And finally proposing the ability for FileSystem browser to create New Folders via context menu.

**Comment by vnen** [comment]
> You can clearly see what a tab does without the title. While it may look nice is a waste of space.

Yes, but it makes hard to direct someone with words. "Click on the Inspector..." "Wait, what's the 'Inspector'?"

I'm all up for removing unnecessary things, but they have really to be unnecessary.

**Comment by alelepd** [comment]
Hey @toger5, thats really cool! :D
Actually, the @trollworkout ideas are very good. This have been done is other apps and it could be implemented in many ways. Blender for example uses lateral tabs to organize panels in some cases. Cinema 4D is smarter, it allows you to display the tabs in different position, top (by default), right and left (like blender), it is also posible to drop the panel tab titles, which is quite handy once you now how everything works.

Flexible panels are very useful.

On the other side, I feel that the nodes design should be more prominent, right now they re transparent, only the tabs are visible.
The texture thumbnails could be a bit different too, it lacks visual order.

@Geequlim nice theme!! 💃

**Comment by trollworkout** [comment]
@vnen  Click on the inspector. OK. What's the inspector? Is it the 3D/2D/Script window nope. Is it the FileBrowser nope. Is it the Animation, Debug, Output. What's left? Where I add nodes and on the right hand side where I set parameters. 

You underestimate people and their ability to figure things out. While you wanna help users and make the UI more friendly you don't wanna spoon feed them things they don't need.

**Comment by alelepd** [comment]
@trollworkout has a good point. Tabs are useful in many cases. For example if you want to make two panels, you'll probably want to use tabs to identity the panels and switch from one to another. Tabs are less useful when there is only one.

Also, tabs are not always need, especially when the content itself is self explanatory (like the 3d view) or the -in some apps- the coordinates panel.

A good UI should be smarter and consider how and when you need it to avoid confusing.

**Comment by trollworkout** [comment]
@alelepd  Of course I didn't mention this above but when panels are merged tabs show up but otherwise they are hidden by default. Perhaps a custom setting like "hide tabs on non-merged panels" could provide a middle ground.

Also the ability to control panel padding would be great.

**Comment by volzhs** [comment]
I agree with @vnen. without title, it could be called as many different name for the same panel.

**Comment by DriNeo** [comment]
@trollworkout  Thanks for the idea but I wonder how do you select a tab if there are two of them on the same dock ? It's useful to save space for the viewport, especially on laptops.

**Comment by trollworkout** [comment]
@DriNeo  I was thinking tabs show up when you have panels merged. The hidden tabs are only for 1 panel alone

**Comment by Elisto** [comment]
wow

**Comment by ssavalot** [comment]
Hi folks

![screen_animation](https://cloud.githubusercontent.com/assets/11367422/16099841/f0cc2a66-335a-11e6-8050-d190b8bf2ed8.png)

I made some updates for flat UI design and I test it with latest Godot alpha 2.1 version. 

Download:
https://github.com/ssavalot/godot_flat_ui

Cheers

**Comment by alelepd** [comment]
Nice one! 👍

**Comment by neikeq** [comment]
@ssavalot Your theme looks great! LineEdits look a bit weird though :P

**Comment by trollworkout** [comment]
The drop boxes and text boxes looks pretty odd almost like  labels.  Still pretty good effort 👍

**Comment by Geequlim** [comment]
I add custom theme support for godot editor in this pr #5243 it allows to load themes generated by the editor as the editor theme.

The following theme is inspired by the [arc-theme](https://github.com/horst3180/arc-theme) GTK theme.

![](https://cdn.rawgit.com/Geequlim/depot/master/images/godot/arc_light1.png)

**Comment by MarianoGnu** [comment]
@Geequlim too brigth! I'm gonna get blind! Turn it off! Turn it OFF!!

**Comment by Geequlim** [comment]
@MarianoGnu There is a dark version https://github.com/Geequlim/godot-themes
Not finished as I`m not good at this stuff.

**Comment by akien-mga** [comment]
Clearing milestone as it's an ongoing discussion, not really something that should be implemented for a specific version.

**Comment by MarianoGnu** [comment]
Btw, will 2.1 change the default theme before stable?

**Comment by reduz** [comment]
no, don't think we'll do a major theme change until 3.0

**Comment by reduz** [comment]
new theme has to be done from scratch to work with HiDPI also, it will be
quite some work

**Comment by krisleesc** [comment]
@reduz Will the new theme be like the original proposal in this thread, or do you guys have something else in mind? I know you've mentioned something about the original designer doing something about the logo too. Any words on any of these two questions?

**Comment by reduz** [comment]
Nothing in mind, maybe yes, won't bother thinking about it until 3.0

**Comment by krisleesc** [comment]
Hey @reduz Thought I'd show you this logo I've been messing with, posted on Facebook too just in case you're on one site more than the other. 
![godot](https://cloud.githubusercontent.com/assets/20977490/17633715/c4fc99c4-609c-11e6-98b8-b68928d16179.jpg)

**Comment by Geequlim** [comment]
@MarianoGnu It was turned off now :smile: 
![](https://cdn.rawgit.com/Geequlim/depot/master/images/godot/arc_dark_LDPI.png)

You can get it form https://github.com/Geequlim/godot-themes

**Comment by Geequlim** [comment]
There is also a light one.
![](https://cdn.rawgit.com/Geequlim/depot/master/images/godot/arc_light_HiDPI.png)

**Comment by akien-mga** [comment]
Looks nice :) I guess you're aware that the inspector section headers ("Sprite", "Node2D", "Transform", etc.) are quite hard to read. Is it because they can't be changed with the current API, or do you still have to do it?

**Comment by Geequlim** [comment]
@akien-mga As mentioned in #6430, the colors in `Editor` section below  cannot be saved.
![qq20160909-0](https://cloud.githubusercontent.com/assets/6964556/18386981/70b4f92e-76cc-11e6-82f9-26b13153e5fc.png)

**Comment by alelepd** [comment]
Hmm, It would be great if it can be changed to white (or very light grey).
The highlight blue over the grey icon on the filesystem also have bad color contrast. A fix could be done this way: using alpha instead of solid grey. Not sure if that's possible though.

**Comment by MarianoGnu** [comment]
@Geequlim i installed and setted it up, but most of the panels haven't changed it's look, just the fonts and colors.


![](https://i.gyazo.com/f67b24a9f4bd84c35095d6e2211b64d2.png)

**Comment by MarianoGnu** [comment]
I found out the reason, looks like all the resource paths are absolute to the scene root instead of local the the resource, this forces you to save the files in the same directory locations that you used to create the theme


![](https://i.gyazo.com/e38e80facde2d914c658b5711c06f9f7.png)

**Comment by Geequlim** [comment]
@MarianoGnu You should run the install.py

**Comment by toger5** [comment]
For windows the theme did not install correct. So i got the same result than MarianoGnu.
Now it works. So installing the theme is as easy as downloading Geequlim's repo -> running the python file -> go to godot settings and choose the right path for the custom theme.
On windows: C:\Users\your_username\AppData\Roaming\Godot\theme\
On mac & linux: Users/your_username/.godot
restart and you are done...

**Comment by KioriSun** [comment]
@Geequlim 
I Would just like to point out that in a smaller screen 1366x768, the theme takes up too much room. I guess it's all the extra 'padding' that the theme uses.
Here is an example:
![2016-09-11_15-05-27](https://cloud.githubusercontent.com/assets/10463922/18419460/bf3c87e0-7831-11e6-9a79-9b6727f5eac8.png)

For comparison, the original theme:
![godot windows opt tools 32_2016-09-11_15-06-11](https://cloud.githubusercontent.com/assets/10463922/18419464/d6f88488-7831-11e6-9717-b2546ae8c163.png)
 
Overall much more functional. ;)

**Comment by Geequlim** [comment]
@KioriSun kiYou can change the margin and font size of the theme for your screen.

**Comment by alelepd** [comment]
Related to that; I always though that these window buttons are too small;
they should be bigger because they represent the main actions that you are
going to take.

2016-09-11 14:30 GMT-04:00 Geequlim <notifications@github.com>:

**Comment by RobotUnderscore** [comment]
Just found this today and it looks absolutely brilliant. I want this implemented badly, new logo and all.

**Comment by Geequlim** [comment]
Hey guys~
Anothor editor theme looks like adobe photoshop is coming
![](https://cdn.rawgit.com/Geequlim/depot/master/images/godot/godot%20adobe%20theme%402x.png)

**Comment by Geequlim** [comment]
@KioriSun This problem should be fixed in latest commit :)

**Comment by MakeGodotGreatAgain** [comment]
why would you ever want to change the logo?

**Comment by ObaniGemini** [comment]
I don't get it. Is there an official theme support or are those only patches replacing files that you have to recompile ?

**Comment by toger5** [comment]
@ObaniGemini there is an official theme support. It still has it's edges but basically it works. I think Geequilim is the only one actually understand how it works and using it (of course there are some more understanding it... ). Most other design previews you see are based on changing textures, some style information in the default_theme.cpp file and recompile.

**Comment by zicklag** [comment]
I just found this and I really enjoy @alelepd's original concept along with the comments about it that I added a :+1:. I has a _very_ professional feel like Unity and Blender does.

I've been keeping up with the master branch and they have been working on a new default theme, but it is mostly just style changes. Was the original concept just an image put together with an image editor or was there actual code for building that UI? I'm really just wondering if that UI is something that can get merged into the Engine or if it is just a concept that would need to be coded into existence.

**Comment by toger5** [comment]
@zicklag it is a concept (made in image manipulating software)
the current theme implementations are influenced by it and a good mix of: possible to execute (in terms of how much work it is) and best looking. In addition to that the new theme will allow for much easier customisation.
Icon change happend already. It is nicer (more modern) but godot won't loose it's branding. (I like it).
-> it can't get merged exactly like proposed. but the new theme will be great you will see!

**Comment by zicklag** [comment]
@toger5, thanks for clearing that up for me. Can't wait to watch this play out. :)

**Comment by toger5** [comment]
@akien-mga since the new ui is pretty close to the concept and most points which got discussed here are solved in a really nice manner right now we could consider closing.

**Comment by akien-mga** [comment]
Indeed :)

**Comment by alelepd** [comment]
Hi everyone! it's been a while since I published this initial design concept. I feel so happy that some of these ideas actually made it to the final release of Godot 3.0 UI.

Also happy with the improvements in other design aspects of Godot, like the new web design.

It was great to see how many of you took the concept and implemented for real, and even expanded to many other sections of the editor.

There is still a lot to do in design for Godot, but this is definitely a step forward!

**Comment by groud** [comment]
@alelepd honestly, your first design is so great I wanted to implement it. I'll give it a try someday ^^

**Comment by reduz** [comment]
Your design was a big inspiration for us. We couldn't follow it more closely because we needed to keep things functional, bit there are still many ideas on it we can explore in the future!

**Comment by aldebaran798** [comment]
We need this! :sweat_smile:

**Comment by APakrohk** [comment]
> @alelepd & @reduz ~ Some ideas I have in mind...
> 
> ![proposal](https://cloud.githubusercontent.com/assets/3820189/8424508/1eaf3a00-1f2e-11e5-9ea4-8a66a969d887.jpg)

> Your design was a big inspiration for us. We couldn't follow it more closely because we needed to keep things functional, bit there are still many ideas on it we can explore in the future!

In version 4.0, how close is Godot to the new design?
Is it associated with a design like version 3?
The Godot interface is in dire need of updates!

**Comment by Calinou** [comment]
@APakrohk The editor theme will probably be redesigned in Godot 4.0. I have a [WIP branch](https://github.com/Calinou/godot/tree/improve-editor-theme) for this, but it needs to be rebased.

**Comment by toger5** [comment]
@Calinou can we get screenshots 😍

**Comment by alelepd** [comment]
@APakrohk Thanks for checking. Actually, I have many new ideas for a new UI (and website) design but I haven't got enough time to work on it. I'll try to make some progress soon.

**Comment by Calinou** [comment]
@toger5 I posted some on IRC recently, but I don't have access to it right now.
